### ISSUE TITLE
vivbin failes to run GUI

### ISSUE THREAD

**Issue by SantasNOPsled** [vivisect/vivisect#540]
I am unable to launch the GUI with the command vivbin PATH_TO_EXE. Vivbin hangs after the analysis is complete. I used  pip install "vivisect[gui]" to setup the program.
pip produces the following output:

Collecting vivisect[gui]
  Using cached vivisect-1.0.8-py3-none-any.whl (4.2 MB)
Requirement already satisfied: pyasn1<0.5.0,>=0.4.5 in /home/kos/.local/lib/python3.10/site-packages (from vivisect[gui]) (0.4.8)
Requirement already satisfied: pyasn1-modules<0.3.0,>=0.2.4 in /home/kos/.local/lib/python3.10/site-packages (from vivisect[gui]) (0.2.8)
Requirement already satisfied: cxxfilt<0.3.0,>=0.2.1 in /home/kos/.local/lib/python3.10/site-packages (from vivisect[gui]) (0.2.2)
Requirement already satisfied: msgpack<1.1.0,>=1.0.0 in /usr/lib/python3.10/site-packages (from vivisect[gui]) (1.0.3)
Requirement already satisfied: pycparser>=2.20 in /usr/lib/python3.10/site-packages (from vivisect[gui]) (2.21)
Requirement already satisfied: pyqt5==5.15.1 in /home/kos/.local/lib/python3.10/site-packages (from vivisect[gui]) (5.15.1)
Requirement already satisfied: pyqtwebengine==5.15.1 in /home/kos/.local/lib/python3.10/site-packages (from vivisect[gui]) (5.15.1)
Requirement already satisfied: PyQt5-sip<13,>=12.8 in /usr/lib/python3.10/site-packages (from pyqt5==5.15.1->vivisect[gui]) (12.11.0)
Installing collected packages: vivisect
Successfully installed vivisect-1.0.8

I've tested vivisect on Pop_OS and Manjaro and observed the same behavior.

**Comment by SantasNOPsled** [comment]
I've tracked down the following issue in my system logs:
systemd-coredump[62801]: [🡕] Process 62796 (QtWebEngineProc) of user 1000 dumped core.

Module /home/kos/.local/lib/python3.10/site-packages/PyQt5/Qt/libexec/QtWebEngineProcess with build-id 8783aff35ab85ce0d6348de67e15cc8b20b6bcf6

**Comment by SantasNOPsled** [comment]
I've gotten the GUI to run in Ubuntu following the directions for vivisectION located here: https://github.com/atlas0fd00m/VivisectION

If anyone has any insights on why it's failing in other environments I'd appreciate hearing about it.

**Comment by rakuy0** [comment]
I pulled down a manjaro ISO last night. Seems like there's some QT callbacks that aren't firing, Specifically this isn't completing: https://github.com/vivisect/vivisect/blob/master/envi/qt/memcanvas.py#L61, and the runJavascript callbacks aren't firing either. Without those, nothing in the UI will work. And also the loadFinished callback isn't firing here: https://github.com/vivisect/vivisect/blob/master/envi/qt/memcanvas.py#L60. So things aren't loading or firing, but it's only on Manjaro, which is bizarre. I tired a clean install on ubuntu, and it all seemed fine and worked as intended from just the normal pip install. 

Concerning that loop.exec: It was a hack I had to put in during the pyqt4->pyqt5 migration in order to deal with pyqt5's async (but not native python async, but QT's multiprocessed in house async). I needed it at the time in order to make all the pieces that rely on the memcanvas have a guarantee that the entire page is actually loaded and not have to rewrite the entire UI subsystem. Without that forced synchronization, a bunch of other things don't work properly. 

Side note: removing that loop.exec does actually cause the main UI to show up, but important features like the memory view or funcgraph view don't render anything at all.

I'll poke more at this during the week. But thankfully it seems like this is constrained to Manjaro. I'm hoping the fix is just "decorate that thing with a qt decorator to denote what to do with it", but I have no guarantees on that.

But thank you for bringing this up! It's definitely reproducible and weird. Hopefully it's easily fixable too.
